src/Composer/Util/Platform.php

```diff
@@ -173,6 +173,53 @@ public static function isWindows(): bool
         return \defined('PHP_WINDOWS_VERSION_BUILD');
     }
 
+    public static function isDocker(): bool
+    {
+        if (null !== self::$isDocker) {
+            return self::$isDocker;
+        }
+
+        // cannot check so assume no
+        if ((bool) ini_get('open_basedir')) {
+            return self::$isDocker = false;
+        }
+
+        // .dockerenv and .containerenv are present in some cases but not reliably
+        if (!file_exists('/.dockerenv') || file_exists('/run/.containerenv') || file_exists('/var/run/.containerenv')) {
```

Comment:
Is the leading bang (`!`) here correct? This looks like it will short-circuit too soon. On my computer, I get this `if` block returning `isDocker()` when I'm not running in Docker at all.

```suggestion
        if (file_exists('/.dockerenv') || file_exists('/run/.containerenv') || file_exists('/var/run/.containerenv')) {
```

Reply:
You're right that's a mistake.